blocks/main_loop.py

```diff
@@ -231,6 +231,12 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                if other not in ['before_batch', 'after_batch']:
+                    raise ValueError("I'm being compared to something \
+                                      I don't know!")
+
         for extension in self.extensions:
             extension.dispatch(method_name, *args)
```

Comment:
`method_name` here is a string, which should now be a custom `Callback` instance e.g. use `Callback(method_name)`